src/assembly/extension.xml

```diff
@@ -0,0 +1,22 @@
+<assembly xmlns="http://maven.apache.org/ASSEMBLY/2.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.0.0 http://maven.apache.org/xsd/assembly-2.0.0.xsd">
+    <id>openrefine-commons-${openrefine.version}</id>
+    <includeBaseDirectory>false</includeBaseDirectory>
+    <formats>
+        <format>zip</format>
+    </formats>
+    <fileSets>
+        <fileSet>
+            <directory>module/</directory>
```

Comment:
My fileSets look like this now... because I saw that `module/MOD-INF/lib` is where dependencies are getting put and with the sample template as is now... it wasn't picking up the greyed out `lib` folder?

1. why is the module/MOD-INF/lib put into `.gitignore` ?
2. Then I'm wondering if we should add the some of the below also, like in my case the `.pom` file from groovy-all getting pulled in by maven dependencies plugin
```
    <fileSets>
        <fileSet>
            <directory>module/</directory>
            <outputDirectory>module/</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>module/MOD-INF/lib/</directory>
            <outputDirectory>module/MOD-INF/lib</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
                <exclude>**/*.pom</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>src/main/resources/module</directory>
            <outputDirectory>module</outputDirectory>
        </fileSet>
    </fileSets>
```

Reply:
OK, the above isn't quite working... hmm... the ZIP file ends up being the top-level module/ directory, rather than something like `groovy/` .  when I extract the `refine-groovy-1.0.zip` and look at it...

![image](https://github.com/OpenRefine/sample-extension/assets/986438/85bb3d16-e60c-42c5-a111-22325007d128)
![image](https://github.com/OpenRefine/sample-extension/assets/986438/58585008-1001-458e-bcf5-10849ec0cc9f)

Reply:
ah, `modules/` is actually stuff only for `core` ?  And extensions should assemble and then output into their own extension name top-level folder..   So...I think it needs to look like this?

```
    <fileSets>
        <fileSet>
            <directory>module/</directory>
            <outputDirectory>sample-extension/</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>module/MOD-INF/lib/</directory>
            <outputDirectory>sample-extension/MOD-INF/lib</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
                <exclude>**/*.pom</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>src/main/resources/module</directory>
            <outputDirectory>sample-extension</outputDirectory>
        </fileSet>
    </fileSets>
    ```

Reply:
I think the current behaviour is the desired one, no? Does it not work when you extract the zip file into the extensions directory?

Reply:
I extract the `refine-groovy-1.0.zip` into the `webapp/extensions` folder and it creates a `module/` folder.  Not anything like a `refine-groovy` folder or just simply `groovy` folder.

![image](https://github.com/OpenRefine/sample-extension/assets/986438/94c8a472-aa57-4a4d-a013-c6ee077cbeb4)

Reply:
If I change the `<outputDirectory>` for the assembly, then things inside the ZIP look better, I think?

```
    <fileSets>
        <fileSet>
            <directory>module/</directory>
            <outputDirectory>refine-groovy/</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>module/MOD-INF/lib/</directory>
            <outputDirectory>refine-groovy/MOD-INF/lib</outputDirectory>
            <excludes>
                <exclude>**/*.java</exclude>
                <exclude>**/*.pom</exclude>
            </excludes>
        </fileSet>
        <fileSet>
            <directory>src/main/resources/module</directory>
            <outputDirectory>refine-groovy</outputDirectory>
        </fileSet>
    </fileSets>
    ```
Using above it creates a ZIP like this now:
![image](https://github.com/OpenRefine/sample-extension/assets/986438/3bec9d50-4337-419f-912d-1c20657d02e9)

Reply:
Nevermind.  Fixed!  See https://github.com/OpenRefine/sample-extension/pull/9/files#r1615171588